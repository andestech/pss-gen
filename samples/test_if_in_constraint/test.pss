component pss_top {

	action root_a {
		rand string instn;
		rand int lmul;

        constraint instn in ["vadd_vv", "vsub_vv", "vnsrl_wv", "vnsrl_wv"];

		constraint {
			if (instn in ["vnsrl_wv", "vnsrl_wv"]) lmul == 4;
			else lmul == 8;
		}
		
        exec body ASM =
		"""instn={{instn}}, lmul={{lmul}}""";

	}
}

