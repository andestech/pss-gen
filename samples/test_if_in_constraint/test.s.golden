instn=vsub_vv, lmul=8
